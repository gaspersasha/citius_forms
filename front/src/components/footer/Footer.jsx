import React from 'react';
import { useRouter } from 'next/router';
import { CONTACTS } from '~constants';
import { Trustpilot, NavigationLink, configTrustpilot } from '~components';
import { Logo } from '~assets/svg';
import { links, footerSocials } from './helpers';
import s from './footer.module.sass';

const { PHONE, EMAIL, PHONE_LINES_OPEN_SCHEDULE } = CONTACTS;

const Footer = () => {
  const { pathname } = useRouter();
  const isHomePage = pathname === '/';

  const socialsList = footerSocials.map((item) => (
    <a
      href={item.href}
      className={s[item.title]}
      rel={item.rel}
      target={item.target}
      key={item.title}
    >
      <span hidden>{item.title}</span>
    </a>
  ));

  const linksList = links.map((link) => (
    <li key={link.title} className={s.siteLinksListItem}>
      {link.target ? (
        <a className={s.siteLink} href={link.href} target={link.target}>
          {link.title}
        </a>
      ) : (
        <NavigationLink absolute className={s.siteLink} href={link.href}>
          {link.title}
        </NavigationLink>
      )}
    </li>
  ));

  const phoneFormatter = (phone) => {
    if (phone && typeof phone === 'string') {
      const arrPhone = phone.split('');

      arrPhone.splice(4, 0, ' ');
      arrPhone.splice(8, 0, ' ');

      return arrPhone.join('');
    }
  };

  return (
    <footer className={s.footer}>
      <div className={s.topBlock}>
        <div className={s.topBlockContainer}>
          <div className={s.logoLinkContainer}>
            <NavigationLink absolute href="/" className={s.logoLink}>
              <Logo className={s.logo} />
            </NavigationLink>
          </div>
          <div className={s.contacts}>
            <a className={s.phone} href={`tel:${PHONE}`}>
              {phoneFormatter(PHONE)}
            </a>
            <a className={s.emailLink} href={`mailto:${EMAIL}`}>
              {EMAIL}
            </a>
            <p>{PHONE_LINES_OPEN_SCHEDULE}</p>
          </div>
          <div className={s.socials}>{socialsList}</div>
        </div>
      </div>
      <div className={s.mainBlock}>
        <div className={s.mainBlockContainer}>
          <ul className={s.siteLinksList}>{linksList}</ul>
          <div className={s.socialsSmallScreens}>{socialsList}</div>
          <div className={s.tpWrapper}>
            <Trustpilot
              options={configTrustpilot({
                theme: 'light',
              })}
            />
          </div>
        </div>
      </div>
      <div className={s.copyright}>
        <div className={s.copyrightContainer}>
          {isHomePage && (
            <img
              className={s.bvrlaLogo}
              src="/images/bvrla-logo.png"
              alt="BVRLA"
            />
          )}
          <p>
            2002-2021 Buyacar Ltd t/a &quot;Buyacar&quot; - no part of this
            website may be reproduced without prior consent. Taking or use of
            images or text is a breach of copyright.&nbsp;
            <a href="/cars/291/buyacar-terms-and-conditions">
              Terms&nbsp;And&nbsp;Conditions
            </a>
          </p>
          <p>
            Buyacar Limited t/a Buyacar; is authorised and regulated by the
            Financial Conduct Authority (667368). We act as a credit broker not
            a lender. We can introduce you to a limited number of lenders who
            may be able to offer you finance facilities for your purchase. We
            will only introduce you to these lenders. We may receive a
            commission payment from the finance provider if you decide to enter
            into an agreement with them. The nature of this commission is as
            follows: we receive a commission based on a percentage of the total
            amount of finance taken by the customer. The commission received
            does not affect the amount you will pay under your finance
            agreement. You can request for us to disclose the amount of any
            commission received. You may be able to obtain finance for your
            purchase from other lenders and you are encouraged to seek
            alternative quotations. If you would like to know how we handle
            complaints, please ask for a copy of our complaints handling
            process. You can also find information about referring a complaint
            to the Financial Ombudsman Service (FOS) at&nbsp;
            <a
              href="https://financial-ombudsman.org.uk"
              target="_blank"
              rel="noreferrer"
            >
              financial-ombudsman.org.uk
            </a>
            .
          </p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
