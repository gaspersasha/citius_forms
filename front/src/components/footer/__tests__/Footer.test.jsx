import React from 'react';
import { useRouter } from 'next/router';

import Footer from '../Footer';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

useRouter.mockImplementation(() => ({
  pathname: '/',
}));

describe('<Footer />', () => {
  const wrapper = shallow(<Footer isEmpty={false} />);

  it('rendered without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
