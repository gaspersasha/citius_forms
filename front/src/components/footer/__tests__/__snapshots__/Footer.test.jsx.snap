// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> rendered without errors 1`] = `
<footer
  className="footer"
>
  <div
    className="topBlock"
  >
    <div
      className="topBlockContainer"
    >
      <div
        className="logoLinkContainer"
      >
        <NavigationLink
          absolute={true}
          activeClassName=""
          activeTowardsRoot={false}
          className="logoLink"
          href="/"
          onClick={null}
        >
          <Logo
            className="logo"
            viewBox="0 0 195 57.1"
            xmlns="http://www.w3.org/2000/svg"
          />
        </NavigationLink>
      </div>
      <div
        className="contacts"
      >
        <a
          className="phone"
          href="tel:666"
        >
          666  
        </a>
        <a
          className="emailLink"
          href="mailto:hello@email.co.uk"
        >
          hello@email.co.uk
        </a>
        <p>
          Phone lines open Mon to Thurs, 9am-7pm; Fri to Sat, 9am-5pm; Sun Closed
        </p>
      </div>
      <div
        className="socials"
      >
        <a
          className="facebook"
          href="https://www.facebook.com/buyacarcouk-118472818183789"
          key="facebook"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            facebook
          </span>
        </a>
        <a
          className="twitter"
          href="https://twitter.com/buy_a_car"
          key="twitter"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            twitter
          </span>
        </a>
        <a
          className="linkedin"
          href="https://www.linkedin.com/company/buyacar-co-uk"
          key="linkedin"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            linkedin
          </span>
        </a>
        <a
          className="email"
          href="mailto:hello@email.co.uk"
          key="email"
        >
          <span
            hidden={true}
          >
            email
          </span>
        </a>
        <a
          className="instagram"
          href="https://www.instagram.com/buyacaruk/"
          key="instagram"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            instagram
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    className="mainBlock"
  >
    <div
      className="mainBlockContainer"
    >
      <ul
        className="siteLinksList"
      >
        <li
          className="siteLinksListItem"
          key="Reservation"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/cars"
            onClick={null}
          >
            Reservation
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Used Cars"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/cars"
            onClick={null}
          >
            Used Cars
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="New Cars in Stock"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/cars/new-cars"
            onClick={null}
          >
            New Cars in Stock
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Vans"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/vans"
            onClick={null}
          >
            Vans
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="How It Works"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/how-it-works"
            onClick={null}
          >
            How It Works
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="FAQ"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/how-it-works/faq"
            onClick={null}
          >
            FAQ
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="About Us"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/how-it-works/about"
            onClick={null}
          >
            About Us
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="As seen in the press"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/press"
            onClick={null}
          >
            As seen in the press
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Buyer's Guides"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/buyers-guides"
            onClick={null}
          >
            Buyer's Guides
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="News"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/news"
            onClick={null}
          >
            News
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Advice"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/advice"
            onClick={null}
          >
            Advice
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Complaints"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/how-it-works/contact-us#complaints"
            onClick={null}
          >
            Complaints
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Initial Disclosure Document"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/cars/1224/initial-disclosure-document"
            onClick={null}
          >
            Initial Disclosure Document
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Work For Us"
        >
          <a
            className="siteLink"
            href="https://www.autovia.co.uk/careers/search/?department=&brand=BuyaCar&location=&typeOfEmployment="
            target="_blank"
          >
            Work For Us
          </a>
        </li>
        <li
          className="siteLinksListItem"
          key="Dealer Sales"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/dealer-sales"
            onClick={null}
          >
            Dealer Sales
          </NavigationLink>
        </li>
        <li
          className="siteLinksListItem"
          key="Privacy Policy"
        >
          <a
            className="siteLink"
            href="https://www.autovia.co.uk/privacy-policy"
            target="_blank"
          >
            Privacy Policy
          </a>
        </li>
        <li
          className="siteLinksListItem"
          key="Cookie Policy"
        >
          <a
            className="siteLink"
            href="https://www.autovia.co.uk/cookies-policy"
            target="_blank"
          >
            Cookie Policy
          </a>
        </li>
        <li
          className="siteLinksListItem"
          key="Sitemap"
        >
          <NavigationLink
            absolute={true}
            activeClassName=""
            activeTowardsRoot={false}
            className="siteLink"
            href="/sitemap"
            onClick={null}
          >
            Sitemap
          </NavigationLink>
        </li>
      </ul>
      <div
        className="socialsSmallScreens"
      >
        <a
          className="facebook"
          href="https://www.facebook.com/buyacarcouk-118472818183789"
          key="facebook"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            facebook
          </span>
        </a>
        <a
          className="twitter"
          href="https://twitter.com/buy_a_car"
          key="twitter"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            twitter
          </span>
        </a>
        <a
          className="linkedin"
          href="https://www.linkedin.com/company/buyacar-co-uk"
          key="linkedin"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            linkedin
          </span>
        </a>
        <a
          className="email"
          href="mailto:hello@email.co.uk"
          key="email"
        >
          <span
            hidden={true}
          >
            email
          </span>
        </a>
        <a
          className="instagram"
          href="https://www.instagram.com/buyacaruk/"
          key="instagram"
          rel="noreferrer"
          target="_blank"
        >
          <span
            hidden={true}
          >
            instagram
          </span>
        </a>
      </div>
      <div
        className="tpWrapper"
      >
        <Trustpilot
          options={
            Object {
              "accountLink": "https://uk.trustpilot.com/review/website",
              "businessUnitId": "4bdcd9ad000064000505fa5a",
              "className": "trustpilot-widget",
              "locale": "en-GB",
              "scriptPath": "https://widget.trustpilot.com/bootstrap/v5/tp.widget.bootstrap.min.js",
              "stars": "4,5",
              "styleHeight": "",
              "styleWidth": "100%",
              "templateId": "53aa8807dec7e10d38f59f32",
              "theme": "light",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="copyright"
  >
    <div
      className="copyrightContainer"
    >
      <img
        alt="BVRLA"
        className="bvrlaLogo"
        src="/images/bvrla-logo.png"
      />
      <p>
        2002-2021 Buyacar Ltd t/a "Buyacar" - no part of this website may be reproduced without prior consent. Taking or use of images or text is a breach of copyright. 
        <a
          href="/cars/291/buyacar-terms-and-conditions"
        >
          Terms And Conditions
        </a>
      </p>
      <p>
        Buyacar Limited t/a Buyacar; is authorised and regulated by the Financial Conduct Authority (667368). We act as a credit broker not a lender. We can introduce you to a limited number of lenders who may be able to offer you finance facilities for your purchase. We will only introduce you to these lenders. We may receive a commission payment from the finance provider if you decide to enter into an agreement with them. The nature of this commission is as follows: we receive a commission based on a percentage of the total amount of finance taken by the customer. The commission received does not affect the amount you will pay under your finance agreement. You can request for us to disclose the amount of any commission received. You may be able to obtain finance for your purchase from other lenders and you are encouraged to seek alternative quotations. If you would like to know how we handle complaints, please ask for a copy of our complaints handling process. You can also find information about referring a complaint to the Financial Ombudsman Service (FOS) at 
        <a
          href="https://financial-ombudsman.org.uk"
          rel="noreferrer"
          target="_blank"
        >
          financial-ombudsman.org.uk
        </a>
        .
      </p>
    </div>
  </div>
</footer>
`;
